test/equality_test.dart

```diff
@@ -175,6 +175,25 @@ main() {
     expect(equality.hash("foo"), isNot(equals(equality.hash("bar"))));
     expect(equality.hash("fÕÕ"), isNot(equals(equality.hash("fõõ"))));
   });
+
+  test("EqualityBy", () {
+    var equality = new EqualityBy<NonCanonicalElement, String>((e) => e.id);
```

Comment:
I'd probably avoid defining a custom class here and just have this be `new EqualityBy<List<String>, String>((e) => e.first)` (or `new EqualityBy((List<String> e) => e.first)`).